### DISCUSSION TITLE
MAVLink

### DISCUSSION THREAD

**Discussion by xxJohnxx** [iNavFlight/inav#6635] [General]
Hi,

I recently started to play around with a little project utilizing MAVLink telemetry together with iNav. Initially I was just poking around with the available options, however one way or another I found myself working on a little GCS project. 
The current project goal is not to re-create a fully featured, extensive mission planning suite like MissionPlanner, but rather a easy to use, lightweight tool to get some more Telemetry options for iNav rover/airplane/quadcopter builds.

After getting most of the data-downlink set up already, I started to poke around with the available options for uplink. While the iNav wiki here mentions, that there isn‘t any uplink functionality I still figured out that an upload is possible using the Mission_Count/Mission_Request sequence. This however, seems only possible when iNav is not armed. Also there seems to be no way that iNav can receive the Mission_Set_Current/Mission_Current command chain. Both of those likely are necessary to be able to re-program a mission underway (be that a rover or an airbone vehicle). 

Now to my question: Are there any plans for an implementation of such futures? After reading up on discussions I am unsure to what extent MAVLink will be implemented into iNav. Can anyone shed some information about that? Is it even a goal for the iNav development or am I missing the point?

Thank you and best regards,

Jonathan

**Comment by lida2003** [comment]
Well, long time ago (Feb 24, 2021)

I'm interested in MAVLink also. Maybe ArduPilot is much better to use MAVLink~~~
